src/core/mount.c

```diff
@@ -1592,11 +1603,51 @@ static int mount_get_timeout(Unit *u, usec_t *timeout) {
         return 1;
 }
 
+static void synthesize_root_mount(Manager *m) {
```

Comment:
i figure synthesize_root_mount() could return an int, indeed. will fix.

mount_enumerate() is a vtable function however. It definitely makes sense to make it return an int too, but that would mean we'd have to change the unit vtable first, and alter all unit type implementations. Makes a lot of sense, but is more invasive, hence let's leave that for a later PR...